core/trino-main/src/main/java/io/trino/sql/planner/AnonymizePlanRewriter.java

```diff
@@ -0,0 +1,785 @@
+            }
+
+            anonymizedSymbols.put(anonymize(key), anonymizedValue);
+        });
+        return anonymizedSymbols.build();
+    }
+
+    private IndexHandle anonymize(IndexHandle handle)
+    {
+        return new IndexHandle(
+                handle.getCatalogName(),
+                new ConnectorTransactionHandle() {},
+                new ConnectorIndexHandle() {});
+    }
+
+    private TableHandle anonymize(TableHandle handle)
+    {
+        TableSchema tableSchema = metadata.getTableSchema(session, handle);
+        return new TableHandle(
+                handle.getCatalogName(),
+                new AnonymizedTableHandle(
+                        new QualifiedObjectName(
+                                anonymize(tableSchema.getQualifiedName().getCatalogName()),
+                                anonymize(tableSchema.getQualifiedName().getSchemaName()),
+                                anonymize(tableSchema.getQualifiedName().getObjectName())),
+                        handle.getConnectorHandle().getClass().getName()),
+                new ConnectorTransactionHandle() {});
+    }
+
+    private Aggregation anonymize(Aggregation aggregation)
+    {
+        return new Aggregation(
+                aggregation.getResolvedFunction(),
+                anonymizeExpressions(aggregation.getArguments()),
+                aggregation.isDistinct(),
+                aggregation.getFilter().map(this::anonymize),
+                aggregation.getOrderingScheme().map(this::anonymize),
+                aggregation.getMask().map(this::anonymize));
+    }
+
+    private OrderingScheme anonymize(OrderingScheme scheme)
+    {
+        return new OrderingScheme(
+                anonymize(scheme.getOrderBy()),
+                anonymize(scheme.getOrderings()));
+    }
+
+    private ListMultimap<Symbol, Symbol> anonymize(ListMultimap<Symbol, Symbol> symbols)
+    {
+        ImmutableListMultimap.Builder<Symbol, Symbol> anonymizedSymbols = ImmutableListMultimap.builder();
+        symbols.forEach((key, val) -> anonymizedSymbols.put(anonymize(key), anonymize(val)));
+        return anonymizedSymbols.build();
+    }
+
+    private List<Symbol> anonymize(List<Symbol> symbols)
+    {
+        return symbols.stream()
+                .map(this::anonymize)
+                .collect(toImmutableList());
+    }
+
+    private Set<Symbol> anonymize(Set<Symbol> symbols)
+    {
+        return symbols.stream()
+                .map(this::anonymize)
+                .collect(toImmutableSet());
+    }
+
+    private List<String> anonymizeNames(List<String> names)
+    {
+        return names.stream()
+                .map(this::anonymize)
+                .collect(toImmutableList());
+    }
+
+    private Symbol anonymize(Symbol symbol)
+    {
+        return new Symbol(anonymize(symbol.getName()));
+    }
+
+    private String anonymize(String name)
+    {
+        return anonymizedSymbols.computeIfAbsent(name, key -> String.valueOf(name.hashCode()));
```

Comment:
Let's also add a prefix of what was anonymised (table, column, symbol, literal etc.) before the hash so that it's easier to read the anonymised output